subsys/net/lib/mqtt/mqtt_pkt.c

```diff
@@ -815,7 +815,7 @@ int mqtt_unpack_suback(uint8_t *buf, uint16_t length, uint16_t *pkt_id,
 	*pkt_id = ntohs(val_u16);
 	offset += PACKET_ID_SIZE;
 
-	*items = length - offset;
+	*items = rlen - PACKET_ID_SIZE;
```

Comment:
Hah nice catch.   I wrote the patch after some initial debugging (a different way) and then adjusted it later once we realized the actual cause.  Patch message didn't get that change.